htdocs/expedition/card.php

```diff
@@ -362,6 +363,30 @@
 					$qty = "qtyl".$i.'_'.$j;
 				}
 			} else {
+				$p = new Product($db);
+				$res = $p->fetch($objectsrc->lines[$i]->fk_product);
+				if ($res > 0) {
+					if (GETPOST('entrepot_id', 'int') == -1) {
+						$qty .= '_'.$j;
+					}
+
+					if ($p->stockable_product == Product::DISABLED_STOCK) {
```

Comment:
This part of code from line 366 to 389 looks strange:
Currently, default is to have all product managed as stockable.
So we should have code added to avoid to do something when product is not stock managed. But here yo add code to do more things when stock is not managed. 

I think we should not have this part of code from line 366 to 389 and we should have the code to do or not action depending of stockable status somewhere later.
If we remove this part of code, where does the program hangs ?

Reply:
You are right. I changed the test in Expedition::addline() that forbid to add a line if warehouse was empty. You can see it in commit 2407fe070da8d79a9952e0e0a8a0de2423fe57ef

In case of product that need a batch or a serial, stock management seems necessary.